app/view_objects/home_view.rb

```diff
@@ -28,35 +24,52 @@ def draft_apply_trainees_count
 
   attr_reader :trainees
 
-  def populate_state_counts!
-    defaults = Trainee.states.keys.index_with { 0 }
-    counts = trainees.group(:state).count.reverse_merge(defaults)
-
-    if eyts_trainees? == qts_trainees?
-      counts["awarded"] ||= 0
-      counts["recommended_for_award"] ||= 0
-    elsif eyts_trainees?
-      awarded = counts.delete("awarded")
-      counts["eyts_awarded"] = awarded
-
-      recommended = counts.delete("recommended_for_award")
-      counts["eyts_recommended"] = recommended
-    elsif qts_trainees?
-      awarded = counts.delete("awarded")
-      counts["qts_awarded"] = awarded
-
-      recommended = counts.delete("recommended_for_award")
-      counts["qts_recommended"] = recommended
-    end
+  def create_badges
+    @badges = [
+      Badge.new(
+        :in_training,
+        trainees.in_training.count,
+        trainees_path(status: %w[in_training]),
+      ),
+
+      Badge.new(
+        :awarded_this_year,
+        trainees.awarded_in_cycle(current_academic_cycle).count,
+        trainees_path(
+          status: %w[awarded],
+          end_year: "#{current_academic_cycle.start_year} to #{current_academic_cycle.end_year}",
+        ),
+      ),
+
+      Badge.new(
+        :deferred,
+        trainees.deferred.count,
+        trainees_path(status: %w[deferred]),
+      ),
 
-    @state_counts = counts
+      Badge.new(
```

Comment:
I'm wondering, will this incomplete filter only include registered trainees? Trying to have a look in the code although it's not super clear to me